utils/chunked_vector.hh

```diff
@@ -146,11 +142,24 @@ public:
     }
     template <typename... Args>
     T& emplace_back(Args&&... args) {
-        reserve_for_push_back();
-        auto& ret = *new (addr(_size)) T(std::forward<Args>(args)...);
+        pointer ret;
+        if (_capacity > _size) [[likely]] {
```

Comment:
This [[likely]] is wrong. Let's say you have a chunked_vector and you only push back once. Only the unlikely path is taken!

likely/unlikely should be used when there's an overwhelming trend, like do-once initialization or error paths.

Reply:
But if you keeping pushing elements, the way we're dynamically reserving capacity, it is likely that _capacity will ne greater than _size.  Is optimizing (or not pessimizing) chunked_vectors of size 1 that important?
If so, maybe we should have internal capacity like small_vector for the first insertion(s).

Reply:
likely isn't for something being a little bit more likely than another, it's for overwhelming difference in probability.

An in fact it's perfectly possible for only the unlikely path to be ever taken, in some workloads.